pkg/plugins/config/config.go

```diff
@@ -26,7 +26,8 @@ type Cfg struct {
 	AWSExternalId           string
 
 	// Azure Cloud settings
-	Azure *azsettings.AzureSettings
+	Azure            *azsettings.AzureSettings
+	AzureAuthEnabled bool
```

Comment:
just curious why these two are different fields

Reply:
I concur, there's no reason we shouldn't add this field to the `azure` section of the config. As long as we clarify that it's specific to Prometheus it should be fine.

Reply:
For now, it is only for Prometheus. I don't see/find any other place we use it. In the future, if we need it in elsewhere then we can move this in azsettings.